docs/_templates/autosummary/flax_module.rst

```diff
@@ -0,0 +1,23 @@
+{{ fullname | escape | underline}}
+
+.. currentmodule:: {{ module }}
```

Comment:
Is this really necessary? Is it because RTD doesn't have built-in support for dataclasses? Or that it doesn't know that this is a dataclass?

Reply:
What do you mean by "this"? This file or this line?

Reply:
This file is there so we can customize how we want our Flax modules to look, because right now they look really bad.

Consider the following example for `flax.linen.Conv` (but it applied to all Flax Modules):

- BEFORE: https://flax.readthedocs.io/en/latest/_autosummary/flax.linen.Conv.html
- AFTER: 

You see the following things are improved:

- Nicer formatting of parameters summary view. Less whitespace
- Nicer formatting of parameters details. Types are displayed more concise.
- Do not show a docstring for `__init__`
- Do show a docstring for `__call__`
- In the "methods" section in the end, only show the members that aren't inherited, which makes the list much shorter.
- Don't show an `attributes` section.

All these things are done by creating a custom template for our Modules.

Reply:
This is there to make sure we output the Module docstring correctly, see PR description.